evalml/tests/automl_tests/test_automl.py

```diff
@@ -2314,38 +2322,37 @@ def test_automl_respects_pipeline_parameters_with_duplicate_components(mock_scor
     pipeline_linear = BinaryClassificationPipeline(component_graph_linear)
 
     automl = AutoMLSearch(X, y, problem_type="binary", allowed_pipelines=[pipeline_linear, pipeline_dict],
-                          pipeline_parameters={"One Hot Encoder": {"top_n": Categorical([15])},
-                                               "One Hot Encoder_1": {"top_n": Categorical([25])}},
+                          pipeline_parameters={"One Hot Encoder": {"top_n": 15},
+                                               "One Hot Encoder_1": {"top_n": 25}},
                           max_batches=3)
     automl.search()
-    for i, row in automl.full_rankings.iterrows():
-        if "Mode Baseline Binary" in row['pipeline_name']:
-            continue
-        assert row["parameters"]["One Hot Encoder"]["top_n"] == 15
-        assert row["parameters"]["One Hot Encoder_1"]["top_n"] == 25
+    for row in automl.full_rankings.iloc[1:3].parameters:
```

Comment:
Since we'll be separating the logic, `pipeline_parameters` will be responsible for the first batch and as previously it would have represented the hyperparameters as well, now `custom_hyperparameters` will be responsible for that instead. So in this test only the initial batch is being set with these values, while the remaining batches revert to defaults in the case of `One Hot Encoder` (since `top_n` isn't a default hyperparameter range) or start iterating over the hyperparameter ranges in the case of `Random Forest Classifier`.

Reply:
Thanks for explaining! We can preserve the previous behavior by setting `custom_hyperparameters`! I think we should list this as a breaking change. Not sure how many users were actively setting `pipeline_parameters` before, but might be good to give them a heads up to use `custom_hyperparameters` instead.

Reply:
Most definitely, I'll list this as a breaking change!